### ISSUE TITLE
Hovering over super-imposed pts (only) considers hoverinfo of top trace

### ISSUE THREAD

**Issue by bxlxux** [plotly/plotly.js#720]
Hello - I came across this bug https://github.com/plotly/plotly.js/issues/493 though I'm not sure it's the same issue.

I have this test codepen running with "hovermode":"closest" set. The text displays when hovering over the lines. However, when I change the cdn to 1.10.1 or later the hover no longer works. I can only get the text to display when removing "hovermode":"closest", but it's displaying all text based upon x and not the way I'd like it to which is "closest".

[http://codepen.io/bxlxux/pen/jALWAK](url)

**Comment by etpinard** [comment]
Thanks for the report.

Here's a more compact reproducible example: http://codepen.io/etpinard/pen/ZOJQyk?

To note, the problem is also present on `master` and seems to be only affecting scattergeo traces with mode set to `lines`. Changing the issue title accordingly.

**Comment by etpinard** [comment]
Oh wait. My mistake your example isn't a `scattergeo` graph.

The problem is in the cartesian hover code.

**Preliminary diagnostic:** Hovering of superimposed picks the last trace and honours the `hoverinfo` setting present in that last trace. For example, if the last trace has `hoverinfo: 'none'` then no hover labels are shown.

See https://github.com/plotly/plotly.js/pull/438, https://github.com/plotly/plotly.js/pull/495 and https://github.com/plotly/plotly.js/issues/678 for more info on the topic.

Thanks again for this report. In the meantime here's an easy workaround: http://codepen.io/etpinard/pen/qNXNwz 

where simply reordering the traces gets you back to the desired behaviour.

**Comment by etpinard** [comment]
More generally,

 _Should hovering over super-imposed traces on `hovermode: 'closest'` show multiple hover labels_?

**Comment by etpinard** [comment]
Ok. I found a way to fix this (see [`closest-hoverinfo-none`](https://github.com/plotly/plotly.js/compare/closest-hoverinfo-none)) but I'm not convinced it is the way to go.

At the moment, hovering with `hovermode: 'closest'` over super-imposed points picks out the most above point - regardless of its `hoverinfo` value. The same point is used to dispatch `plotly_click` and `plotly_hover` info - which _a priori_ should not care about the `hoverinfo` value.

So, in order to make the _picked_ point consistent across hover label, `plotly_click` and `plotly_hover`, I propose in https://github.com/plotly/plotly.js/issues/678 to add a  `hoverinfo` value to distinguish between traces that should be only hidden on hover (but can still trigger `plotly_click` and `plotly_hover`) and traces that should be completely ignored by the picking routine.

**Comment by bxlxux** [comment]
Thanks for the quick reply! Unfortunately my javascript skills are limited so I may not be able to meaningfully contribute. I did take the javascript from your codepen (http://codepen.io/etpinard/pen/qNXNwz) and add that to my original pen (http://codepen.io/bxlxux/pen/jALWAK). 

I understand the simplification, but since we're running a batch process to generate over 100+ charts our json gets written into a script tag within the html file rather than a url variable you used. That's why it's quite lengthy. 

So if you looked at my pen [http://codepen.io/bxlxux/pen/jALWAK,](url) how could I point to the json? I think this works if I can get the that sorted. Thanks!

**Comment by etpinard** [comment]
I'm not sure I understand your question.

All you need to get back the desired behavior is to reorder your traces such that the above superimposed trace does not have `hoverinfo: 'none'`.

**Comment by bxlxux** [comment]
Okay I see what you're saying. I guess the change in 1.10.1 requires us to do this reordering of traces that we didn't have to do in 1.10.0. We manually changed the hoverinfo to none for several of the objects in our map like the points indicating the city, because we added them to the tooltip on hover. It would be nice to not have to reorder the traces since we have over 100+, but alternatively we can stick with 1.10.0.
